dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/data/DefaultQueryValidator.java

```diff
@@ -147,7 +147,8 @@ else if ( params.hasStartEndDate() && !params.getReportingRates().isEmpty() )
         {
             error = new ErrorMessage( ErrorCode.E7107 );
         }
-        else if ( !params.getFilterIndicators().isEmpty() && params.getFilterOptions( DATA_X_DIM_ID ).size() > 1 )
+        else if ( (!params.getFilterIndicators().isEmpty() || !params.getFilterProgramIndicators().isEmpty())
```

Comment:
Could we please add a unit test to verify the behavior in `DefaultQueryValidator`?

Reply:
Done